carbonplan_trace/v1/landsat_preprocess.py

```
@@ -0,0 +1,329 @@
+    
+    Returns
+    -------
+    xarray dataset
+        Dataset of dimensions x, y, bands (corresponding to bands_of_interest)
+    '''
+
+    if type(item) == str:
+        url_list = [item+'_{}.TIF'.format(band) for band in bands_of_interest]
+    else:
+        url_list = [fix_link(item._stac_obj.assets['{}.TIF'.format(band)]['href']) for band in bands_of_interest]
+    da_list = []
+    for url in url_list:
+        da_list.append(rioxarray.open_rasterio(url, chunks={'x': 1024,
+                                                    'y': 1024}))
+
+    # combine into one dataset
+    ds = xr.concat(da_list, dim='band').to_dataset(dim='band').rename({1: 'reflectance'})
+    ds = ds.assign_coords({'band': bands_of_interest})
+    # fill value is 0; let's switch it to nan
+    ds = ds.where(ds != 0)  
+    ds = ds.where(cog_mask<2)
+    ds = ds.reflectance.to_dataset(dim='band').drop('spatial_ref')
+    ds.attrs["utm_zone_number"] = utm_zone
+    ds.attrs["utm_zone_letter"] = utm_letter
+    ds = calc_NDVI(ds)
+    ds = calc_NDII(ds)
+    return ds.load()
+
+
+def average_stack_of_scenes(ds_list):
+    '''
+    Average across scenes. This will work the same regardless 
+    of whether your scenes are perfectly overlapping or they're offset.
+    However, if they're offset it requires a merge and so the entire 
+    datacube (pre-collapsing) will be instantiated and might make 
+    your kernel explode.
+
+    Parameters
+    ----------
+    ds_list : list
+        List of xarray datsets to average.
+    
+    Returns
+    -------
+    xarray dataset
+        Dataset of dimensions x, y, bands (corresponding to bands_of_interest)
+    '''
+    utm_zone = []
+    utm_letter = []
+    for ds in ds_list:
+        utm_zone.append(ds.attrs['utm_zone_number'])
+        utm_letter.append(ds.attrs['utm_zone_letter'])
+    if (len(set(utm_zone))>1) or (len(set(utm_letter))>1):
+        print('WATCH OUT: youre averaging scenes from multiple utm projections!!')
```

hmm do we need to throw an exception here instead of just a warning message?